### ISSUE TITLE
cuQuantum Python v22.03.0: `ModuleNotFoundError: No module named 'typing_extensions'`

### ISSUE THREAD

**Issue by leofang** [NVIDIA/cuQuantum#3]
### Workaround: please install `typing_extensions` via pip or conda:
```
pip install typing_extensions
```
or 
```
conda install -c conda-forge typing_extensions
```

### Symptom:
```
leof:~$ python
Python 3.9.10 | packaged by conda-forge | (main, Feb  1 2022, 21:24:37) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cuquantum
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/leof/miniforge3/envs/ppqwqq/lib/python3.9/site-packages/cuquantum/__init__.py", line 2, in <module>
    from cuquantum import cutensornet
  File "/home/leof/miniforge3/envs/ppqwqq/lib/python3.9/site-packages/cuquantum/cutensornet/__init__.py", line 2, in <module>
    from cuquantum.cutensornet.configuration import *
  File "/home/leof/miniforge3/envs/ppqwqq/lib/python3.9/site-packages/cuquantum/cutensornet/configuration.py", line 19, in <module>
    from .memory import BaseCUDAMemoryManager
  File "/home/leof/miniforge3/envs/ppqwqq/lib/python3.9/site-packages/cuquantum/cutensornet/memory.py", line 7, in <module>
    from typing_extensions import Protocol, runtime_checkable
ModuleNotFoundError: No module named 'typing_extensions'
```

**Comment by kcgthb** [comment]
Confirmed issue, and workaround, thanks! :+1:

**Comment by leofang** [comment]
Updated docs and README.

**Comment by leofang** [comment]
Unpinning this issue as it's fixed in 22.05:
https://docs.nvidia.com/cuda/cuquantum/python/release_notes.html#cuquantum-python-v22-05-0
